src/subtype.c

```diff
@@ -2346,6 +2366,10 @@ static jl_value_t *intersect_var(jl_tvar_t *b, jl_value_t *a, jl_stenv_t *e, int
                 ub = (jl_value_t*)b;
             }
         }
+        if (has_free_vararg_length(ub, e) && !has_free_vararg_length(a, e)) {
+            bb->ub = ub;
+            ub = (jl_value_t*)b;
+        }
```

Comment:
should this be moved several lines further (just guarding the assignment to lb)? Right now, it seems that it may try to set `bb->ub = b` (which would be invalid) and then `return b` (which would be less accurate) and skips the `check_unsat_bound` on this Tuple

Reply:
Not sure I get it. `has_free_vararg_length` wont return true if `ub` is a typevar, so we' never perform `bb->ub = b` here.
But yes, `check_unsat_bound` should not be skipped.